pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -916,3 +918,103 @@ func MakeNextPodFunc(queue SchedulingQueue) func() *v1.Pod {
 func podInfoKeyFunc(obj interface{}) (string, error) {
 	return cache.MetaNamespaceKeyFunc(obj.(*podInfo).pod)
 }
+
```

Comment:
IMO, this piece of code and its functionality is independent enough that can go to its own file. We can have a `pod_backoff.go` file in the `queue` directory.